### ISSUE TITLE
Cannot set property regexLikeCss of #<Object> which has only a getter

### ISSUE THREAD

**Issue by hqwlkj** [elado/next-with-less#34]
![image](https://user-images.githubusercontent.com/12181423/234817646-4f2e8b89-8778-475b-9d31-c97b9016171d.png)

```
yarn list v1.22.19
warning Filtering by arguments is deprecated. Please use the pattern option instead.
└─ next@13.3.1
✨  Done in 0.72s.

```

**Comment by bingoYB** [comment]
+ 1

**Comment by viztor** [comment]
#32

**Comment by johnmagbag1995** [comment]
Same issue

"next": "13.4.0",
"antd": "^4.24.8",
"next-compose-plugins": "^2.2.1",

**Comment by LeonardoVizzotto** [comment]
I'm facing the same problem with `"next": "^13.4.1"`.

**Comment by AntiMoron** [comment]
Same issue.

**Comment by elado** [comment]
Hi all. I was able to repro this but unfortunately couldn't find a proper solution yet.

The monkeypatch is supposed to allow support for image URLs inside less files. E.g. `background-image: url('./image.png')`.
Looks like Node is blocking writing into that module. something changed about Next.js transpilation in the recent version.
I tried using `Object.defineProperty`, `Reflect.set`, `Reflect.defineProperty` but they either don't work (the property isn't applied) or the error mentioned is thrown.

[This PR](https://github.com/elado/next-with-less/pull/32) is not fixing it - `obj = { ...obj, prop: 'val' }` doesn't mutate the object, but the intention is to mutate.

At this point, I see the following options:
1. Give up on `url` support in less files (least favorite option)
2. Find a JS solution that allows rewriting the export - I'm open to ideas
3. Use a different patching mechanism, like overwriting `Module.prototype.require` (definitely not a fan of this but it might be the only options)

**Comment by AntiMoron** [comment]
https://github.com/vercel/next.js/discussions/49665

**Comment by Mnigos** [comment]
same issue here

**Comment by elado** [comment]
Released a fix in v3.0.0

**Comment by gabek** [comment]
Thank you!
